source/custom/visa_service.custom.cpp

```diff
@@ -344,7 +351,7 @@ ::grpc::Status VisaService::Read(::grpc::ServerContext* context, const ReadReque
     std::string* buffer = response->mutable_buffer();
     buffer->resize(count);
     ViUInt32 return_count{};
-    auto status = library_->Read(vi, reinterpret_cast<ViByte*>(&buffer[0]), count, &return_count);
+    auto status = library_->Read(vi, get_buffer_data_pointer<std::string>(buffer, count), count, &return_count);
```

Comment:
Right, I thought it would need to make the type explicitly to instantiate the template function. But it turns out that removing the type compiles just fine.

Reply:
commit 5